clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java

```diff
@@ -173,23 +186,52 @@ private String apiVersionToText(ApiVersion apiVersion) {
         }
 
         if (apiKey != null) {
-            Short usableVersion = usableVersions.get(apiKey);
-            if (usableVersion == NODE_TOO_OLD)
+            UsableVersion usableVersion = usableVersions.get(apiKey);
+            if (usableVersion.isTooOld())
                 bld.append(" [unusable: node too old]");
-            else if (usableVersion == NODE_TOO_NEW)
+            else if (usableVersion.isTooNew())
                 bld.append(" [unusable: node too new]");
             else
-                bld.append(" [usable: ").append(usableVersion).append("]");
+                bld.append(" [usable: ").append(usableVersion.defaultVersion).append("]");
         }
         return bld.toString();
     }
 
     public ApiVersion apiVersion(ApiKeys apiKey) {
-        for (ApiVersion nodeApiVersion : nodeApiVersions) {
-            if (nodeApiVersion.apiKey == apiKey.id) {
-                return nodeApiVersion;
-            }
+        for (UsableVersion usableVersion : usableVersions.values()) {
+            if (usableVersion.apiVersion.apiKey == apiKey.id)
+                return usableVersion.apiVersion;
         }
         throw new NoSuchElementException();
     }
+
+    private static class UsableVersion {
+        private static final short NODE_TOO_OLD = (short) -1;
+        private static final short NODE_TOO_NEW = (short) -2;
+
+        private final ApiVersion apiVersion;
+        private final Short defaultVersion;
```

Comment:
I found the term `default` a bit confusing here. This is `usableVersion`, right? I guess you were trying to avoid repeating it since the class name is `UsableVersion`. Could we just call it `value`?